src/integ_test/java/games/strategy/engine/lobby/server/db/UserControllerIntegrationTest.java

```diff
@@ -1,69 +1,52 @@
 import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
 
 import java.sql.Connection;
 import java.sql.ResultSet;
 import java.util.UUID;
+import java.util.function.Function;
 import java.util.function.Supplier;
 
+import javax.annotation.Nullable;
+
 import org.junit.Test;
+import org.mindrot.jbcrypt.BCrypt;
 
+import games.strategy.engine.lobby.server.login.RsaAuthenticator;
 import games.strategy.engine.lobby.server.userDB.DBUser;
 import games.strategy.util.MD5Crypt;
+import games.strategy.util.Util;
 
 public class UserControllerIntegrationTest {
 
-  private static final DbTestConnection POSTGRES =
-      new DbTestConnection(
-          Database::getPostgresConnection,
-          new UserController(Database::getPostgresConnection));
+  private static final Supplier<Connection> connectionSupplier = Database::getPostgresConnection;
+  private final UserController controller = new UserController(connectionSupplier);
 
   @Test
   public void testCreate() throws Exception {
-    testCreate(POSTGRES);
-  }
+    final int startCount = getUserCount();
 
-  private static void testCreate(final DbTestConnection dbTestConnection) throws Exception {
-    final DBUser user = givenUser();
-    final int startCount = getUserCount(dbTestConnection.connectionSupplier);
+    createUserWithMd5CryptHash();
+    assertEquals(getUserCount(), startCount + 1);
 
-    dbTestConnection.controller.createUser(user, new HashedPassword(MD5Crypt.crypt(user.getName())));
-
-    final int endCount = getUserCount(dbTestConnection.connectionSupplier);
-    assertEquals(endCount, startCount + 1);
-  }
-
-  private static DBUser givenUser() {
-    final String name = UUID.randomUUID().toString().substring(0, 20);
-    final String email = name + "@none.none";
-    return new DBUser(
-        new DBUser.UserName(name),
-        new DBUser.UserEmail(email));
+    createUserWithBCryptHash();
+    assertEquals(getUserCount(), startCount + 2);
   }
 
-  private static int getUserCount(final Supplier<Connection> connection) throws Exception {
-    try (final Connection dbConnection = connection.get()) {
-      final String sql = "select count(*) from TA_USERS";
-      final ResultSet rs = dbConnection.createStatement().executeQuery(sql);
-      assertTrue(rs.next());
-      return rs.getInt(1);
-    }
-  }
-
-
   @Test
   public void testGet() throws Exception {
-    testGet(POSTGRES);
-  }
+    final DBUser user = createUserWithMd5CryptHash();
 
-  private static void testGet(final DbTestConnection dbTestConnection) throws Exception {
-    final DBUser user = givenUser();
-    dbTestConnection.controller.createUser(user, new HashedPassword(MD5Crypt.crypt(user.getName())));
-
-    final DBUser loadedUser = dbTestConnection.controller.getUserByName(user.getName());
+    final DBUser loadedUser = controller.getUserByName(user.getName());
 
     assertEquals(loadedUser.getName(), user.getName());
     assertEquals(loadedUser.getEmail(), user.getEmail());
```

Comment:
Note that `DBUser` now implements `equals()`, so, if you wanted to, you could replace these three lines with `assertEquals(user, loadedUser);`.